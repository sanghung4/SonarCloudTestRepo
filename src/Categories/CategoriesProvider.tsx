import { ReactNode, createContext } from 'react';

import { Maybe, Category, useProductCategoriesQuery } from 'generated/graphql';
import { useDomainInfo } from 'hooks/useDomainInfo';

type Categories = Maybe<Array<Maybe<Category>>>;

type CategoriesContextType = {
  categories?: Categories;
  categoriesLoading: boolean;
};

export const CategoriesContext = createContext({
  categoriesLoading: false
} as CategoriesContextType);

type Props = {
  children: ReactNode;
};

function CategoriesProvider(props: Props) {
  /**
   * Data
   */
  const { engine } = useDomainInfo();
  const { data: categoriesData, loading: categoriesLoading } =
    useProductCategoriesQuery({
      skip: !engine,
      variables: { engine }
    });

  return (
    <CategoriesContext.Provider
      value={{
        categories: categoriesData?.productCategories?.categories as Categories,
        categoriesLoading
      }}
    >
      {props.children}
    </CategoriesContext.Provider>
  );
}

export default CategoriesProvider;
