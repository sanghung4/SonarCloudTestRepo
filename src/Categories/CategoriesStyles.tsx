import {
  List,
  ListItemButton,
  ListItemText,
  alpha,
  styled
} from '@dialexa/reece-component-library';

export const ListStyled = styled(List)(({ theme }) => ({
  overflowY: 'scroll',
  [theme.breakpoints.up('sm')]: {
    padding: theme.spacing(1.5, 0),
    borderRight: '1px solid #e4e4e4',
    width: '25%',
    flexShrink: 0
  }
}));

export const ListItemButtonStyled = styled(ListItemButton)(({ theme }) => ({
  [theme.breakpoints.up('sm')]: {
    marginTop: 6,
    marginBottom: 6,
    paddingTop: 4,
    paddingBottom: 4,
    '&.Mui-focusVisible': {
      backgroundColor: alpha(theme.palette.primary02.main, 0.1)
    },
    '&.Mui-selected': {
      backgroundColor: alpha(theme.palette.primary02.main, 0.1),
      borderRight: `5px solid ${theme.palette.primary02.main}`,
      '&:hover': {
        backgroundColor: alpha(theme.palette.primary02.main, 0.1)
      }
    }
  },
  '&.MuiListItemButton-gutters': {
    [theme.breakpoints.up('md')]: {
      paddingLeft: 12,
      paddingRight: 12
    },
    [theme.breakpoints.down('md')]: {
      paddingLeft: 32,
      paddingRight: 32
    }
  }
}));

export const ListItemTextStyled = styled(ListItemText)(({ theme }) => ({
  '& .MuiListItemText-primary': {
    [theme.breakpoints.down('md')]: {
      color: theme.palette.text.primary
    }
  }
}));
