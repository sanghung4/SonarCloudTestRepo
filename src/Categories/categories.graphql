query Categories($erp: String) {
  categories(erp: $erp) {
    mincronCategories
    eclipseCategories
  }
}

query ProductCategories($engine: String!) {
  productCategories(engine: $engine) {
    categories {
      name
      children {
        name
        children {
          name
        }
      }
    }
  }
}
