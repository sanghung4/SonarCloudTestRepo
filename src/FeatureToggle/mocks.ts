import { Feature } from 'generated/graphql';

export const mockFeatures: Feature[] = [
  { id: '1', isEnabled: false, name: 'noooo' },
  { id: '2', isEnabled: true, name: 'fine' },
  { id: '3', isEnabled: true, name: 'test' }
];
