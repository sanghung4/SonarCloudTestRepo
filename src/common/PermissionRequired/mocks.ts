import { UserContextType } from 'AuthProvider';
import { Permission } from 'common/PermissionRequired';

export const authConfigNoPerm = {
  authState: {
    isAuthenticated: true
  },
  profile: {
    isEmployee: true,
    isVerified: true
  } as UserContextType['profile'],
  activeFeatures: ['LISTS']
};
export const authConfig = {
  authState: {
    isAuthenticated: true
  },
  profile: {
    userId: 'test',
    permissions: [Permission.EDIT_LIST, Permission.APPROVE_CART],
    isEmployee: true,
    isVerified: true
  } as UserContextType['profile'],
  activeFeatures: ['LISTS']
};
