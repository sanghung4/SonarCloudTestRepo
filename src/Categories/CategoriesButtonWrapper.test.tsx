import { ReactNode } from 'react';
import { waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import useResizeObserver from 'use-resize-observer';

import CategoriesButtonWrapper from 'Categories/CategoriesButtonWrapper';
import { CategoriesContext } from 'Categories/CategoriesProvider';
import { success } from 'Categories/mocks';
import { Category } from 'generated/graphql';
import { setBreakpoint } from 'test-utils/mockMediaQuery';
import { render } from 'test-utils/TestWrapper';

jest.mock('use-resize-observer', () => ({
  __esModule: true,
  default: jest.fn()
}));

type MockCategoryContextProps = {
  children: ReactNode;
  data?: Category[];
  loading: boolean;
};

const MockCategoriesContext = ({
  children,
  data,
  loading
}: MockCategoryContextProps) => {
  const mockValues = {
    categories: data,
    categoriesLoading: loading
  };
  return (
    <CategoriesContext.Provider value={mockValues}>
      {children}
    </CategoriesContext.Provider>
  );
};

const mockCategoryData = success.result.data?.productCategories
  ?.categories as Category[];

describe('CategoriesButtonWrapper tests', () => {
  beforeEach(() => {
    setBreakpoint('desktop');
    (useResizeObserver as jest.Mock).mockReturnValue({
      ref: null,
      width: 1024
    });
  });

  it('should render correctly', () => {
    const { container, getByTestId } = render(
      <MockCategoriesContext data={mockCategoryData} loading={false}>
        <CategoriesButtonWrapper />
      </MockCategoriesContext>
    );
    const browseProductsButton = getByTestId('browse-products-button');
    userEvent.hover(browseProductsButton);
    expect(getByTestId('category1-0')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should render correctly while loading', () => {
    const { container } = render(
      <MockCategoriesContext data={undefined} loading={true}>
        <CategoriesButtonWrapper />
      </MockCategoriesContext>
    );
    expect(container).toMatchSnapshot();
  });

  it('should link correctly', async () => {
    const { getByTestId } = render(
      <MockCategoriesContext data={mockCategoryData} loading={false}>
        <CategoriesButtonWrapper />
      </MockCategoriesContext>
    );
    userEvent.hover(getByTestId('browse-products-button'));
    await waitFor(() => new Promise((res) => setTimeout(res, 0)));
    const category = getByTestId('category3-4-0');
    expect(category).toHaveAttribute(
      'href',
      '/search?&categories=Faucets%2C%20Fixtures%20%26%20Appliances&categories=Fans%20%26%20Ventilation&categories=TBC'
    );
  });

  it('should update categories when hovering over a category', () => {
    const { getByTestId } = render(
      <MockCategoriesContext data={mockCategoryData} loading={false}>
        <CategoriesButtonWrapper />
      </MockCategoriesContext>
    );
    userEvent.hover(getByTestId('browse-products-button'));
    userEvent.hover(getByTestId('category1-2'));
    const pvcLink = getByTestId('category3-2-0');

    expect(pvcLink.innerHTML).toEqual('PVC Pipe &amp; Fittings');
  });
});
