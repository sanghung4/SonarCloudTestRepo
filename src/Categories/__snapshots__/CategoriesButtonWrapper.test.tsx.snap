// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesButtonWrapper tests should render correctly 1`] = `
<div>
  <div
    class="MuiBox-root css-1c3h18e"
  >
    <div
      class="MuiBox-root css-ntca38"
    />
    <button
      class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-9tnmny-MuiButtonBase-root-MuiButton-root"
      data-testid="browse-products-button"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiBox-root css-10rvbm3"
      >
        Browse Products
      </span>
      <span
        class="MuiButton-endIcon MuiButton-iconSizeMedium css-1f03u4k-MuiButton-endIcon"
      >
        <svg
          class="MuiBox-root css-1avs10n"
          sx="[object Object]"
        >
          arrow-drop-down.svg
        </svg>
      </span>
    </button>
    <div
      data-popper-escaped=""
      data-popper-placement="bottom-start"
      data-popper-reference-hidden=""
      id="categories-listing"
      role="tooltip"
      style="position: absolute; top: 0px; left: 0px; z-index: 1300; margin: 0px; transform: translate(0px, 8px);"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiCard-root css-ept2uu-MuiPaper-root-MuiCard-root"
        data-testid="categories-listing"
        style="opacity: 1; transform: scale(1, 1); transform-origin: 0 0 0; transition: opacity 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      >
        <div
          class="MuiBox-root css-s88hxx"
        >
          <ul
            class="MuiList-root MuiBox-root css-qshlp8-MuiList-root"
          >
            <div
              class="MuiListItemButton-root MuiListItemButton-dense MuiListItemButton-gutters Mui-selected MuiButtonBase-root css-1ej0ht7-MuiButtonBase-root-MuiListItemButton-root"
              data-testid="category1-0"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemText-root MuiListItemText-dense css-1y7mhd5-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-caption css-ohe2uh-MuiTypography-root"
                >
                  Faucets, Fixtures & Appliances
                </span>
              </div>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </div>
            <div
              class="MuiListItemButton-root MuiListItemButton-dense MuiListItemButton-gutters MuiButtonBase-root css-1ej0ht7-MuiButtonBase-root-MuiListItemButton-root"
              data-testid="category1-1"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemText-root MuiListItemText-dense css-1y7mhd5-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-caption css-ohe2uh-MuiTypography-root"
                >
                  Plumbing Installation, Tools, Hardware & Safety
                </span>
              </div>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </div>
            <div
              class="MuiListItemButton-root MuiListItemButton-dense MuiListItemButton-gutters MuiButtonBase-root css-1ej0ht7-MuiButtonBase-root-MuiListItemButton-root"
              data-testid="category1-2"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemText-root MuiListItemText-dense css-1y7mhd5-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-caption css-ohe2uh-MuiTypography-root"
                >
                  Pipe & Fittings
                </span>
              </div>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </div>
            <div
              class="MuiListItemButton-root MuiListItemButton-dense MuiListItemButton-gutters MuiButtonBase-root css-1ej0ht7-MuiButtonBase-root-MuiListItemButton-root"
              data-testid="category1-3"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemText-root MuiListItemText-dense css-1y7mhd5-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-caption css-ohe2uh-MuiTypography-root"
                >
                  Heating & Cooling
                </span>
              </div>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </div>
            <div
              class="MuiListItemButton-root MuiListItemButton-dense MuiListItemButton-gutters MuiButtonBase-root css-1ej0ht7-MuiButtonBase-root-MuiListItemButton-root"
              data-testid="category1-4"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemText-root MuiListItemText-dense css-1y7mhd5-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-caption css-ohe2uh-MuiTypography-root"
                >
                  Hot Water, Valves, Irrigation & Pumps
                </span>
              </div>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </div>
            <div
              class="MuiListItemButton-root MuiListItemButton-dense MuiListItemButton-gutters MuiButtonBase-root css-1ej0ht7-MuiButtonBase-root-MuiListItemButton-root"
              data-testid="category1-5"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemText-root MuiListItemText-dense css-1y7mhd5-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-caption css-ohe2uh-MuiTypography-root"
                >
                  TBC
                </span>
              </div>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </div>
            <div
              class="MuiListItemButton-root MuiListItemButton-dense MuiListItemButton-gutters MuiButtonBase-root css-1ej0ht7-MuiButtonBase-root-MuiListItemButton-root"
              data-testid="category1-6"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemText-root MuiListItemText-dense css-1y7mhd5-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-caption css-ohe2uh-MuiTypography-root"
                >
                  Faucets, Fixtures Appliances
                </span>
              </div>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </div>
            <div
              class="MuiListItemButton-root MuiListItemButton-dense MuiListItemButton-gutters MuiButtonBase-root css-1ej0ht7-MuiButtonBase-root-MuiListItemButton-root"
              data-testid="category1-7"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemText-root MuiListItemText-dense css-1y7mhd5-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-caption css-ohe2uh-MuiTypography-root"
                >
                  Faucets, FiXtures & Appliances
                </span>
              </div>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </div>
          </ul>
          <div
            class="MuiBox-root css-t23v9o"
          >
            <div
              class="MuiBox-root css-2ges0f"
            >
              <h5
                class="MuiTypography-root MuiTypography-h5 css-16xaac9-MuiTypography-root"
              >
                Product Categories
                 
                <span
                  class="MuiBox-root css-kaq1dv"
                >
                  (
                  6
                  )
                </span>
              </h5>
              <a
                class="MuiTypography-root MuiTypography-body2 MuiLink-root MuiLink-underlineNone css-1yuky57-MuiTypography-root-MuiLink-root"
                href="/search?categories=Faucets%2C%20Fixtures%20%26%20Appliances"
              >
                View All
              </a>
            </div>
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-1cyz8w0-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-3 css-1uf91w7-MuiGrid-root"
              >
                <a
                  class="MuiTypography-root MuiTypography-body2 MuiLink-root MuiLink-underlineNone css-1dw1a9l-MuiTypography-root-MuiLink-root"
                  data-testid="category2-0"
                  href="/search?&categories=Faucets%2C%20Fixtures%20%26%20Appliances&categories=TBC"
                >
                  TBC
                </a>
                <a
                  class="MuiTypography-root MuiTypography-caption MuiLink-root MuiLink-underlineNone css-1djh7kt-MuiTypography-root-MuiLink-root"
                  data-testid="category3-0-0"
                  href="/search?&categories=Faucets%2C%20Fixtures%20%26%20Appliances&categories=TBC&categories=TBC"
                >
                  TBC
                </a>
              </div>
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-3 css-1uf91w7-MuiGrid-root"
              >
                <a
                  class="MuiTypography-root MuiTypography-body2 MuiLink-root MuiLink-underlineNone css-1dw1a9l-MuiTypography-root-MuiLink-root"
                  data-testid="category2-1"
                  href="/search?&categories=Faucets%2C%20Fixtures%20%26%20Appliances&categories=Faucets%2C%20Showers%20%26%20Bathroom%20Accessories"
                >
                  Faucets, Showers & Bathroom Accessories
                </a>
                <a
                  class="MuiTypography-root MuiTypography-caption MuiLink-root MuiLink-underlineNone css-1djh7kt-MuiTypography-root-MuiLink-root"
                  data-testid="category3-1-0"
                  href="/search?&categories=Faucets%2C%20Fixtures%20%26%20Appliances&categories=Faucets%2C%20Showers%20%26%20Bathroom%20Accessories&categories=Showers"
                >
                  Showers
                </a>
                <a
                  class="MuiTypography-root MuiTypography-caption MuiLink-root MuiLink-underlineNone css-1djh7kt-MuiTypography-root-MuiLink-root"
                  data-testid="category3-1-1"
                  href="/search?&categories=Faucets%2C%20Fixtures%20%26%20Appliances&categories=Faucets%2C%20Showers%20%26%20Bathroom%20Accessories&categories=Bathroom%20Accessories"
                >
                  Bathroom Accessories
                </a>
                <a
                  class="MuiTypography-root MuiTypography-caption MuiLink-root MuiLink-underlineNone css-1djh7kt-MuiTypography-root-MuiLink-root"
                  data-testid="category3-1-2"
                  href="/search?&categories=Faucets%2C%20Fixtures%20%26%20Appliances&categories=Faucets%2C%20Showers%20%26%20Bathroom%20Accessories&categories=Tub%20%26%20Shower"
                >
                  Tub & Shower
                </a>
                <a
                  class="MuiTypography-root MuiTypography-caption MuiLink-root MuiLink-underlineNone css-1djh7kt-MuiTypography-root-MuiLink-root"
                  data-testid="category3-1-3"
                  href="/search?&categories=Faucets%2C%20Fixtures%20%26%20Appliances&categories=Faucets%2C%20Showers%20%26%20Bathroom%20Accessories&categories=Bathroom%20Faucets"
                >
                  Bathroom Faucets
                </a>
                <a
                  class="MuiTypography-root MuiTypography-caption MuiLink-root MuiLink-underlineNone css-1djh7kt-MuiTypography-root-MuiLink-root"
                  data-testid="category3-1-4"
                  href="/search?&categories=Faucets%2C%20Fixtures%20%26%20Appliances&categories=Faucets%2C%20Showers%20%26%20Bathroom%20Accessories&categories=Kitchen%20Faucets"
                >
                  Kitchen Faucets
                </a>
                <a
                  class="MuiTypography-root MuiTypography-caption MuiLink-root MuiLink-underlineNone css-1djh7kt-MuiTypography-root-MuiLink-root"
                  data-testid="category3-1-5"
                  href="/search?&categories=Faucets%2C%20Fixtures%20%26%20Appliances&categories=Faucets%2C%20Showers%20%26%20Bathroom%20Accessories&categories=Tub%20Filler"
                >
                  Tub Filler
                </a>
                <a
                  class="MuiTypography-root MuiTypography-caption MuiLink-root MuiLink-underlineNone css-1djh7kt-MuiTypography-root-MuiLink-root"
                  data-testid="category3-1-6"
                  href="/search?&categories=Faucets%2C%20Fixtures%20%26%20Appliances&categories=Faucets%2C%20Showers%20%26%20Bathroom%20Accessories&categories=Spare%20Parts"
                >
                  Spare Parts
                </a>
                <a
                  class="MuiTypography-root MuiTypography-caption MuiLink-root MuiLink-underlineNone css-1djh7kt-MuiTypography-root-MuiLink-root"
                  data-testid="category3-1-7"
                  href="/search?&categories=Faucets%2C%20Fixtures%20%26%20Appliances&categories=Faucets%2C%20Showers%20%26%20Bathroom%20Accessories&categories=TBC"
                >
                  TBC
                </a>
                <a
                  class="MuiTypography-root MuiTypography-caption MuiLink-root MuiLink-underlineNone css-1djh7kt-MuiTypography-root-MuiLink-root"
                  data-testid="category3-1-8"
                  href="/search?&categories=Faucets%2C%20Fixtures%20%26%20Appliances&categories=Faucets%2C%20Showers%20%26%20Bathroom%20Accessories&categories=Kitchen%20Accessories"
                >
                  Kitchen Accessories
                </a>
                <a
                  class="MuiTypography-root MuiTypography-caption MuiLink-root MuiLink-underlineNone css-1djh7kt-MuiTypography-root-MuiLink-root"
                  data-testid="category3-1-9"
                  href="/search?&categories=Faucets%2C%20Fixtures%20%26%20Appliances&categories=Faucets%2C%20Showers%20%26%20Bathroom%20Accessories&categories=CATCLEAN"
                >
                  CATCLEAN
                </a>
              </div>
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-3 css-1uf91w7-MuiGrid-root"
              >
                <a
                  class="MuiTypography-root MuiTypography-body2 MuiLink-root MuiLink-underlineNone css-1dw1a9l-MuiTypography-root-MuiLink-root"
                  data-testid="category2-2"
                  href="/search?&categories=Faucets%2C%20Fixtures%20%26%20Appliances&categories=Appliances"
                >
                  Appliances
                </a>
                <a
                  class="MuiTypography-root MuiTypography-caption MuiLink-root MuiLink-underlineNone css-1djh7kt-MuiTypography-root-MuiLink-root"
                  data-testid="category3-2-0"
                  href="/search?&categories=Faucets%2C%20Fixtures%20%26%20Appliances&categories=Appliances&categories=Refrigerators%20And%20Freezers"
                >
                  Refrigerators And Freezers
                </a>
                <a
                  class="MuiTypography-root MuiTypography-caption MuiLink-root MuiLink-underlineNone css-1djh7kt-MuiTypography-root-MuiLink-root"
                  data-testid="category3-2-1"
                  href="/search?&categories=Faucets%2C%20Fixtures%20%26%20Appliances&categories=Appliances&categories=Hoods%20and%20Vents"
                >
                  Hoods and Vents
                </a>
                <a
                  class="MuiTypography-root MuiTypography-caption MuiLink-root MuiLink-underlineNone css-1djh7kt-MuiTypography-root-MuiLink-root"
                  data-testid="category3-2-2"
                  href="/search?&categories=Faucets%2C%20Fixtures%20%26%20Appliances&categories=Appliances&categories=Outdoor%20Grills%20and%20BBQ"
                >
                  Outdoor Grills and BBQ
                </a>
                <a
                  class="MuiTypography-root MuiTypography-caption MuiLink-root MuiLink-underlineNone css-1djh7kt-MuiTypography-root-MuiLink-root"
                  data-testid="category3-2-3"
                  href="/search?&categories=Faucets%2C%20Fixtures%20%26%20Appliances&categories=Appliances&categories=Ranges%20and%20Stoves"
                >
                  Ranges and Stoves
                </a>
                <a
                  class="MuiTypography-root MuiTypography-caption MuiLink-root MuiLink-underlineNone css-1djh7kt-MuiTypography-root-MuiLink-root"
                  data-testid="category3-2-4"
                  href="/search?&categories=Faucets%2C%20Fixtures%20%26%20Appliances&categories=Appliances&categories=Microwaves"
                >
                  Microwaves
                </a>
                <a
                  class="MuiTypography-root MuiTypography-caption MuiLink-root MuiLink-underlineNone css-1djh7kt-MuiTypography-root-MuiLink-root"
                  data-testid="category3-2-5"
                  href="/search?&categories=Faucets%2C%20Fixtures%20%26%20Appliances&categories=Appliances&categories=Garbage%20Disposals"
                >
                  Garbage Disposals
                </a>
                <a
                  class="MuiTypography-root MuiTypography-caption MuiLink-root MuiLink-underlineNone css-1djh7kt-MuiTypography-root-MuiLink-root"
                  data-testid="category3-2-6"
                  href="/search?&categories=Faucets%2C%20Fixtures%20%26%20Appliances&categories=Appliances&categories=Appliance%20Accessories"
                >
                  Appliance Accessories
                </a>
                <a
                  class="MuiTypography-root MuiTypography-caption MuiLink-root MuiLink-underlineNone css-1djh7kt-MuiTypography-root-MuiLink-root"
                  data-testid="category3-2-7"
                  href="/search?&categories=Faucets%2C%20Fixtures%20%26%20Appliances&categories=Appliances&categories=Dishwashers"
                >
                  Dishwashers
                </a>
                <a
                  class="MuiTypography-root MuiTypography-caption MuiLink-root MuiLink-underlineNone css-1djh7kt-MuiTypography-root-MuiLink-root"
                  data-testid="category3-2-8"
                  href="/search?&categories=Faucets%2C%20Fixtures%20%26%20Appliances&categories=Appliances&categories=Dryers"
                >
                  Dryers
                </a>
                <a
                  class="MuiTypography-root MuiTypography-caption MuiLink-root MuiLink-underlineNone css-1djh7kt-MuiTypography-root-MuiLink-root"
                  data-testid="category3-2-9"
                  href="/search?&categories=Faucets%2C%20Fixtures%20%26%20Appliances&categories=Appliances&categories=Ovens"
                >
                  Ovens
                </a>
              </div>
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-3 css-1uf91w7-MuiGrid-root"
              >
                <a
                  class="MuiTypography-root MuiTypography-body2 MuiLink-root MuiLink-underlineNone css-1dw1a9l-MuiTypography-root-MuiLink-root"
                  data-testid="category2-3"
                  href="/search?&categories=Faucets%2C%20Fixtures%20%26%20Appliances&categories=Toilets%2C%20Urinals%2C%20Sinks%20%26%20Flush%20Valves"
                >
                  Toilets, Urinals, Sinks & Flush Valves
                </a>
                <a
                  class="MuiTypography-root MuiTypography-caption MuiLink-root MuiLink-underlineNone css-1djh7kt-MuiTypography-root-MuiLink-root"
                  data-testid="category3-3-0"
                  href="/search?&categories=Faucets%2C%20Fixtures%20%26%20Appliances&categories=Toilets%2C%20Urinals%2C%20Sinks%20%26%20Flush%20Valves&categories=TBC"
                >
                  TBC
                </a>
                <a
                  class="MuiTypography-root MuiTypography-caption MuiLink-root MuiLink-underlineNone css-1djh7kt-MuiTypography-root-MuiLink-root"
                  data-testid="category3-3-1"
                  href="/search?&categories=Faucets%2C%20Fixtures%20%26%20Appliances&categories=Toilets%2C%20Urinals%2C%20Sinks%20%26%20Flush%20Valves&categories=Toilet%20%26%20Flush%20Valve%20Spare%20Parts"
                >
                  Toilet & Flush Valve Spare Parts
                </a>
                <a
                  class="MuiTypography-root MuiTypography-caption MuiLink-root MuiLink-underlineNone css-1djh7kt-MuiTypography-root-MuiLink-root"
                  data-testid="category3-3-2"
                  href="/search?&categories=Faucets%2C%20Fixtures%20%26%20Appliances&categories=Toilets%2C%20Urinals%2C%20Sinks%20%26%20Flush%20Valves&categories=CATCLEAN"
                >
                  CATCLEAN
                </a>
                <a
                  class="MuiTypography-root MuiTypography-caption MuiLink-root MuiLink-underlineNone css-1djh7kt-MuiTypography-root-MuiLink-root"
                  data-testid="category3-3-3"
                  href="/search?&categories=Faucets%2C%20Fixtures%20%26%20Appliances&categories=Toilets%2C%20Urinals%2C%20Sinks%20%26%20Flush%20Valves&categories=Manual%20Flush%20Valves"
                >
                  Manual Flush Valves
                </a>
                <a
                  class="MuiTypography-root MuiTypography-caption MuiLink-root MuiLink-underlineNone css-1djh7kt-MuiTypography-root-MuiLink-root"
                  data-testid="category3-3-4"
                  href="/search?&categories=Faucets%2C%20Fixtures%20%26%20Appliances&categories=Toilets%2C%20Urinals%2C%20Sinks%20%26%20Flush%20Valves&categories=Spare%20Parts"
                >
                  Spare Parts
                </a>
              </div>
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-3 css-1uf91w7-MuiGrid-root"
              >
                <a
                  class="MuiTypography-root MuiTypography-body2 MuiLink-root MuiLink-underlineNone css-1dw1a9l-MuiTypography-root-MuiLink-root"
                  data-testid="category2-4"
                  href="/search?&categories=Faucets%2C%20Fixtures%20%26%20Appliances&categories=Fans%20%26%20Ventilation"
                >
                  Fans & Ventilation
                </a>
                <a
                  class="MuiTypography-root MuiTypography-caption MuiLink-root MuiLink-underlineNone css-1djh7kt-MuiTypography-root-MuiLink-root"
                  data-testid="category3-4-0"
                  href="/search?&categories=Faucets%2C%20Fixtures%20%26%20Appliances&categories=Fans%20%26%20Ventilation&categories=TBC"
                >
                  TBC
                </a>
              </div>
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-3 css-1uf91w7-MuiGrid-root"
              >
                <a
                  class="MuiTypography-root MuiTypography-body2 MuiLink-root MuiLink-underlineNone css-1dw1a9l-MuiTypography-root-MuiLink-root"
                  data-testid="category2-5"
                  href="/search?&categories=Faucets%2C%20Fixtures%20%26%20Appliances&categories=Tools"
                >
                  Tools
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CategoriesButtonWrapper tests should render correctly while loading 1`] = `
<div>
  <div
    class="MuiBox-root css-1c3h18e"
  >
    <div
      class="MuiBox-root css-ntca38"
    />
    <button
      class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-12ryi9q-MuiButtonBase-root-MuiButton-root"
      data-testid="browse-products-button"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiBox-root css-10rvbm3"
      >
        Browse Products
      </span>
      <span
        class="MuiButton-endIcon MuiButton-iconSizeMedium css-1f03u4k-MuiButton-endIcon"
      >
        <svg
          class="MuiBox-root css-0"
          sx="[object Object]"
        >
          arrow-drop-down.svg
        </svg>
      </span>
    </button>
  </div>
</div>
`;
